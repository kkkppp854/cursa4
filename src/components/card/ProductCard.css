.product-card {
  width: 1.9em;
}
.card-title{
  height: 10vh;}

.product-card:hover .product-image {
  transform: scale(1.1);
}

.product-image-wrapper {
  overflow: hidden;
}

.product-image {
  transition: transform 0.3s ease;
  width: 100%;
  height: auto;
}
