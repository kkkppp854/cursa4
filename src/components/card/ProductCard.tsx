import './ProductCard.css';

interface Products {
  title: string;
  price: number;
  image: string;
}

const ProductCard = ({ title, price, image }: Products) => (
  <div className="card product-card h-100 mx-0">
    <div className="product-image-wrapper">
      <img src={image} alt={title} className="card-img-top product-image" />
    </div>
    <div className="card-body">
      <h5 className="card-title">{title}</h5>
      <div className="d-flex justify-content-between align-items-center pt-2 mt-2">
        <span className='fs-5'>{price} грн</span>
        <a href="#" className="btn btn-info btn-sm px-3">Купити</a>
      </div>
    </div>
  </div>
);

export default ProductCard;