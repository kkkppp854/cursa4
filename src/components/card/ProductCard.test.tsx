import { render } from '@testing-library/react';
import ProductCard from './ProductCard';
import '@testing-library/jest-dom';

const mockProduct = {
  title: "Каблучка з фінітами",
  price: 10000,
  image: "",
  description: "Опис тестового товару"
};

describe('ProductCard', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(<ProductCard {...mockProduct} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
