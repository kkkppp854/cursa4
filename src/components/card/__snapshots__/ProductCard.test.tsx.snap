// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductCard renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card product-card h-100 mx-0"
  >
    <div
      class="product-image-wrapper"
    >
      <img
        alt="Каблучка з фінітами"
        class="card-img-top product-image"
      />
    </div>
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Каблучка з фінітами
      </h5>
      <div
        class="d-flex justify-content-between align-items-center pt-2 mt-2"
      >
        <span
          class="fs-5"
        >
          10000 грн
        </span>
        <a
          class="btn btn-info btn-sm px-3"
          href="#"
        >
          Купити
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
